cmd/res/profiles/camera.yaml

```
@@ -825,3 +825,10 @@ deviceCommands:
     resourceOperations:
       - { deviceResource: "Profiles" }
       - { deviceResource: "StreamUri" }
+  -
+    name: "SnapshotFunction"
+    readWrite: "R"
+    isHidden: false
+    resourceOperations:
+      - { deviceResource: "Profiles" }
+      - { deviceResource: "Snapshot" }
```

This one needs to be `Snapshot` so the `Snapshot` command uses it. Rename the resource `SnapshotFrame` to avoid name confusion.

```suggestion
    name: "Snapshot"
    readWrite: "R"
    isHidden: false
    resourceOperations:
      - { deviceResource: "Profiles" }
      - { deviceResource: "SnapshotFrame" }
```

---

But it seems to be already using it with the existing one only. Need to work with the team if the `deviceResource` name has to be changed as we want to keep them consistent as much as possible.

---

I do not think we need this at all? Does it get the snapshot for all profiles?

---

As in, the `Snapshot` command already works without it.

---

> But it seems to be already using it with the existing one only. Need to work with the team if the `deviceResource` name has to be changed as we want to keep them consistent as much as possible.

Your Snap shot command isn't using this command. It would have to use `SnapshotFunction` as the command name. So to @ajcasagrande's point does it need a Profile??

Also you could just name this one `SnapShot` which hides the resource with the same name. I would be more explicate and set IsHiden: true on the resource

---

@lenny-intel Aren't these just macros to get more than one resource from a single command? If so, I'm not sure how this would even work, or how the streamuri command above would work either, unless EdgeX is smart enough to dig through the first response and use it as input to the second call

---

@ajcasagrande , correct on how they work to combine multiple resources in one command. I also was wondering  how they are working since it just reads both resources. It doesn't pass the Profiles to the other command.

---

Honestly I don't understand this or I didn't have too much time to spend understanding this. I just saw the `videoStream` command and made it like that. If this is wrong my guess is even `VideoStream` needs to be changed. As @ajcasagrande mentioned it might be working without it.

---

@presatish , @ajcasagrande , `videoStream` is not used in the PostMan collection. So never been tested? IMO it should be removed along with `SnapshotFunction`